airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3StorageOperations.java

```diff
@@ -95,23 +95,15 @@ public String getBucketObjectPath(final String namespace, final String streamNam
 
   /**
    * Create a directory object at the specified location. Creates the bucket if necessary.
-   *
-   * @param objectPath The directory to create. Must be a nonempty string.
    */
   @Override
-  public void createBucketObjectIfNotExists(final String objectPath) {
+  public void createBucketIfNotExists() {
     final String bucket = s3Config.getBucketName();
-    final String folderPath = objectPath.endsWith("/") ? objectPath : objectPath + "/";
     if (!doesBucketExist(bucket)) {
       LOGGER.info("Bucket {} does not exist; creating...", bucket);
       s3Client.createBucket(bucket);
       LOGGER.info("Bucket {} has been created.", bucket);
     }
-    if (!s3Client.doesObjectExist(bucket, folderPath)) {
```

Comment:
not sure I understand why we're removing this - do we not need to verify the ability to create a new s3 object?

Reply:
IIUC - S3 (and GCS) have a simulated folder hierarchy, i.e. while are some helpers in the client tooling that simulate listing objects under a certain "directory", these tools are essentially just looking at a string prefix. The removed lines below are creating a zero content object at the `folderPath` which is unnecessary because the a parent path doesn't need to exist before objects are placed since there are no real folders/directories. After removing those lines the `objectPath` parameter is no longer required because its purpose was to create the unnecessary empty object at `folderPath` and the spirit of the method changes to ensuring the bucket exists.

Reply:
ah got it, that makes sense.

also it looks like we check object create+delete in a separate method https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3BaseChecks.java#L36